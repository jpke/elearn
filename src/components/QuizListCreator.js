import React from 'react'
import {Link} from 'react-router'

//creates list of quizzes available to user
//if no quizzes passed in, will return a reminder to select a course
//if admin, displays option to edit quiz
const QuizListCreator = (items, selectFunction, admin, editQuiz) => {
  if(!items) return <li>No quizzes listed <br /> Have you selected a course?</li>;
  if(admin) {
    return items.map((item, index) => {
      let itemName = item.name || item.title;
      return (
        <li key={index} className="listed-item" onClick={() => selectFunction(item)}>
          <div>{itemName}</div>
          <Link to="/quiz/modify"><button onClick={() => {editQuiz("modify")}}>Edit</button></Link>
        </li>
        )
    })
  }
  return items.map((item, index) => {
    let itemName = item.name || item.title;
    return <li key={index} onClick={() => selectFunction(item)}>{itemName}</li>
  })
}

export default QuizListCreator;
