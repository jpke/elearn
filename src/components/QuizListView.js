import React from 'react';
import {Link} from 'react-router';

//view for QuizListContainer
//displays list of quizzes available in selected course
//if no course selected, listCreator returns list item reminding user to select a course
//displays a checkmark next to quizzes listed that user has passed
//offers option to create a new quiz is user is a course admin
const QuizListView = (props) => (
  <div>
    <h3>Select quiz</h3>
    <div className="quiz-list-container">
      <ul className="quiz-list">
        {props.listCreator(props.quizzes, props.selectQuiz, props.admin, props.editQuiz, props.passed)}
      </ul>
      <ul className="passed">
        {props.quizzes && props.passedQuizList(props.quizzes, props.passed)}
      </ul>
    </div>
    {props.admin && <Link to="/quiz/modify"><button className= "create-quiz" onClick={() => {props.editQuiz()}}>Create Quiz</button></Link>}
  </div>
);

export default QuizListView;
