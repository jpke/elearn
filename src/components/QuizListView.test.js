/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import QuizListView from './QuizListView';
import QuizListCreator from '../utils/QuizListCreator';
import passedQuizList from '../utils/passedQuizList';

let quizzes, passed;
beforeEach(() => {
  quizzes = [
    {
      _id: "1111",
      title: "example Quiz"
    },
    {
      _id: "1112",
      title: "example Quiz 2"
    }
  ];

  passed = [{
    _id: "2222",
    of: "1111"
  }];
})

it('renders correctly non admin user who has no listed quizzes', () => {
  quizzes = [];
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={""}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={false}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly admin user who has no listed quizzes', () => {
  quizzes = [];
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={""}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={true}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly non admin user who has passed two of two quizzes', () => {
  passed.push({
    _id: "2223",
    of: "1112"
  });
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={quizzes}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={false}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly admin user who has passed two of two quizzes', () => {
  passed.push({
    _id: "2223",
    of: "1112"
  });
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={quizzes}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={true}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly non admin user who has passed one of two quizzes', () => {
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={quizzes}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={false}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly admin user who has passed one of two quizzes', () => {
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={quizzes}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={true}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly non admin user who has passed zero of two quizzes', () => {
  passed[0].of = "2222"
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={quizzes}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={false}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly admin user who has passed zero of two quizzes', () => {
  passed[0].of = "2222"
  const tree = renderer.create(
    <QuizListView
      listCreator={QuizListCreator}
      quizzes={quizzes}
      passedQuizList={passedQuizList}
      passed={passed}
      admin={true}
      selectQuiz={jest.fn()}
      editQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
