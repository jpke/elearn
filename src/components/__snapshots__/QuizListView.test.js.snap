exports[`test renders correctly admin user who has no listed quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li>
        No quizzes listed 
        <br />
         Have you selected a course?
      </li>
    </ul>
    <ul
      className="passed">
      
    </ul>
  </div>
  <a
    onClick={[Function]}
    style={Object {}}>
    <button
      className="create-quiz"
      onClick={[Function]}>
      Create Quiz
    </button>
  </a>
</div>
`;

exports[`test renders correctly admin user who has passed one of two quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li
        className="listed-item"
        onClick={[Function]}>
        <div>
          example Quiz
        </div>
        <a
          onClick={[Function]}
          style={Object {}}>
          <button
            onClick={[Function]}>
            Edit
          </button>
        </a>
      </li>
      <li
        className="listed-item"
        onClick={[Function]}>
        <div>
          example Quiz 2
        </div>
        <a
          onClick={[Function]}
          style={Object {}}>
          <button
            onClick={[Function]}>
            Edit
          </button>
        </a>
      </li>
    </ul>
    <ul
      className="passed">
      <li>
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24">
          <g>
            <path
              d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
          </g>
        </svg>
      </li>
      <li />
    </ul>
  </div>
  <a
    onClick={[Function]}
    style={Object {}}>
    <button
      className="create-quiz"
      onClick={[Function]}>
      Create Quiz
    </button>
  </a>
</div>
`;

exports[`test renders correctly admin user who has passed two of two quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li
        className="listed-item"
        onClick={[Function]}>
        <div>
          example Quiz
        </div>
        <a
          onClick={[Function]}
          style={Object {}}>
          <button
            onClick={[Function]}>
            Edit
          </button>
        </a>
      </li>
      <li
        className="listed-item"
        onClick={[Function]}>
        <div>
          example Quiz 2
        </div>
        <a
          onClick={[Function]}
          style={Object {}}>
          <button
            onClick={[Function]}>
            Edit
          </button>
        </a>
      </li>
    </ul>
    <ul
      className="passed">
      <li>
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24">
          <g>
            <path
              d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
          </g>
        </svg>
      </li>
      <li>
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24">
          <g>
            <path
              d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
          </g>
        </svg>
      </li>
    </ul>
  </div>
  <a
    onClick={[Function]}
    style={Object {}}>
    <button
      className="create-quiz"
      onClick={[Function]}>
      Create Quiz
    </button>
  </a>
</div>
`;

exports[`test renders correctly admin user who has passed zero of two quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li
        className="listed-item"
        onClick={[Function]}>
        <div>
          example Quiz
        </div>
        <a
          onClick={[Function]}
          style={Object {}}>
          <button
            onClick={[Function]}>
            Edit
          </button>
        </a>
      </li>
      <li
        className="listed-item"
        onClick={[Function]}>
        <div>
          example Quiz 2
        </div>
        <a
          onClick={[Function]}
          style={Object {}}>
          <button
            onClick={[Function]}>
            Edit
          </button>
        </a>
      </li>
    </ul>
    <ul
      className="passed">
      <li />
      <li />
    </ul>
  </div>
  <a
    onClick={[Function]}
    style={Object {}}>
    <button
      className="create-quiz"
      onClick={[Function]}>
      Create Quiz
    </button>
  </a>
</div>
`;

exports[`test renders correctly non admin user who has no listed quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li>
        No quizzes listed 
        <br />
         Have you selected a course?
      </li>
    </ul>
    <ul
      className="passed">
      
    </ul>
  </div>
</div>
`;

exports[`test renders correctly non admin user who has passed one of two quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li
        onClick={[Function]}>
        example Quiz
      </li>
      <li
        onClick={[Function]}>
        example Quiz 2
      </li>
    </ul>
    <ul
      className="passed">
      <li>
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24">
          <g>
            <path
              d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
          </g>
        </svg>
      </li>
      <li />
    </ul>
  </div>
</div>
`;

exports[`test renders correctly non admin user who has passed two of two quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li
        onClick={[Function]}>
        example Quiz
      </li>
      <li
        onClick={[Function]}>
        example Quiz 2
      </li>
    </ul>
    <ul
      className="passed">
      <li>
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24">
          <g>
            <path
              d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
          </g>
        </svg>
      </li>
      <li>
        <svg
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24">
          <g>
            <path
              d="M9 16.17l-4.17-4.17-1.42 1.41 5.59 5.59 12-12-1.41-1.41z" />
          </g>
        </svg>
      </li>
    </ul>
  </div>
</div>
`;

exports[`test renders correctly non admin user who has passed zero of two quizzes 1`] = `
<div>
  <h3>
    Select quiz
  </h3>
  <div
    className="quiz-list-container">
    <ul
      className="quiz-list">
      <li
        onClick={[Function]}>
        example Quiz
      </li>
      <li
        onClick={[Function]}>
        example Quiz 2
      </li>
    </ul>
    <ul
      className="passed">
      <li />
      <li />
    </ul>
  </div>
</div>
`;
